src/BaGetter/ConfigureBaGetterServer.cs

```diff
@@ -0,0 +1,54 @@
+using BaGetter.Core;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Cors.Infrastructure;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.HttpOverrides;
+using Microsoft.Extensions.Options;
+
+namespace BaGetter;
+
+public class ConfigureBaGetterServer
+    : IConfigureOptions<CorsOptions>
+    , IConfigureOptions<FormOptions>
+    , IConfigureOptions<ForwardedHeadersOptions>
+    , IConfigureOptions<IISServerOptions>
+{
+    public const string CorsPolicy = "AllowAll";
+    private readonly BaGetterOptions _baGetterOptions;
+
+    public ConfigureBaGetterServer(IOptions<BaGetterOptions> baGetterOptions)
+    {
+        _baGetterOptions = baGetterOptions.Value;
+    }
+
+
+    public void Configure(CorsOptions options)
+    {
+        // TODO: Consider disabling this on production builds.
+        options.AddPolicy(
+            CorsPolicy,
+            builder => builder.AllowAnyOrigin()
+                .AllowAnyMethod()
+                .AllowAnyHeader());
+    }
+
+    public void Configure(FormOptions options)
+    {
+        // Allow packages up to ~8GiB in size
+        options.MultipartBodyLengthLimit = (long) _baGetterOptions.MaxPackageSizeGiB * int.MaxValue / 2;
+    }
+
+    public void Configure(ForwardedHeadersOptions options)
+    {
+        options.ForwardedHeaders = ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
+
+        // Do not restrict to local network/proxy
+        options.KnownNetworks.Clear();
+        options.KnownProxies.Clear();
+    }
+
+    public void Configure(IISServerOptions options)
+    {
+        options.MaxRequestBodySize = 262144000;
```

Comment:
What about this static variable? Shouldn't it has the same value as `MultipartBodyLengthLimit`?

Reply:
I thought so too, but it wasn't necessary for that 7GB package to be uploaded so I guess it's either not used for multipart payloads, or maybe for each part separately...which would also make it work so not that. Not sure.

Reply:
Wait no, @Schroedingers-Cat used a docker image, which means they didn't test it on IIS, which means this might still break when used on IIS. I'll change it.